consensus/double_sign.go

```diff
@@ -45,7 +45,7 @@ func (consensus *Consensus) checkDoubleSign(recvMsg *FBFTMessage) bool {
 								Msg("could not read shard state")
 							return true
 						}
-						offender := shard.FromLibBLSPublicKeyUnsafe(recvMsg.SenderPubkey)
+						offender := shard.FromLibBLSPublicKeyUnsafe(recvMsg.SenderPubkey.Object)
```

Comment:
leader is going to call `checkDoubleSign` on every commit, should we use the already available bytes in the SenderPubkey? Is unsafe call really required here?

Reply:
Yup, should be optimized. Done.